packages/guides/src/Compiler/NodeTransformers/MenuNodeAddEntryTransformer.php

```diff
@@ -184,4 +172,55 @@ public static function isAbsoluteFile(string $expectedFile): bool
     {
         return str_starts_with($expectedFile, '/');
     }
+
+    /** @param DocumentEntryNode[] $documentEntriesInTree */
+    private function attachDocumentEntriesToParents(
+        array $documentEntriesInTree,
+        CompilerContext $compilerContext,
+        string $currentPath,
+    ): void {
+        foreach ($documentEntriesInTree as $documentEntryInToc) {
+            if ($documentEntryInToc->isRoot() || $currentPath === $documentEntryInToc->getFile()) {
+                // The root page may not be attached to any other
+                continue;
+            }
+
+            if ($documentEntryInToc->getParent() !== null && $documentEntryInToc->getParent() !== $compilerContext->getDocumentNode()->getDocumentEntry()) {
+                $this->logger->warning(sprintf(
+                    'Document %s has been added to parents %s and %s',
+                    $documentEntryInToc->getFile(),
+                    $documentEntryInToc->getParent()->getFile(),
+                    $compilerContext->getDocumentNode()->getDocumentEntry()->getFile(),
+                ));
+            }
+
+            if ($documentEntryInToc->getParent() !== null) {
+                continue;
+            }
+
+            $documentEntryInToc->setParent($compilerContext->getDocumentNode()->getDocumentEntry());
+            $compilerContext->getDocumentNode()->getDocumentEntry()->addChild($documentEntryInToc);
+        }
+    }
+
+    private function addSubSectionsToMenuEntries(DocumentEntryNode $documentEntry, MenuEntryNode $menuEntry): void
+    {
+        foreach ($documentEntry->getSections() as $section) {
```

Comment:
Does a document always have a single main section? I think we should just ignore the first level 1 section. Others should be added.

Reply:
There could be more then one section if two headings of level 1 are used. It is not common or recommended but can happen